nncf/tensorflow/quantization/algorithm.py

```diff
@@ -368,71 +360,73 @@ def _run_batchnorm_adaptation(self, model: tf.keras.Model) -> None:
-
-        if not self.quantize_outputs:
-            for nncf_node in nncf_graph.get_output_nodes():
-                node_key = nncf_graph.get_node_key_by_id(nncf_node.node_id)
-                for quantized_node in self._get_quantized_nodes_for_output(nncf_graph,
-                                                                           insertion_point_node_keys, node_key):
-                    quantized_node_key = nncf_graph.get_node_key_by_id(quantized_node.node_id)
-                    insertion_point_node_keys.remove(quantized_node_key)
-
-        insertion_point_node_keys = [point for point in insertion_point_node_keys
-                                     if _filter_fn(nncf_graph.get_node_by_key(point))]
-        insertion_point_layer_node_names = [nncf_graph.get_node_by_key(ip).node_name
-                                            for ip in insertion_point_node_keys]
-
-        return insertion_point_layer_node_names
-
-    def _get_input_preprocessing_nodes(self, nncf_graph: NNCFGraph, node_key: str,
-                                       preprocessing_nodes: List[NNCFNode] = None) -> List[NNCFNode]:
-        if preprocessing_nodes is None:
-            preprocessing_nodes = []
-
-        node = nncf_graph.get_node_by_key(node_key)
-        successors = nncf_graph.get_next_nodes(node)
-        if len(successors) == 1:
-            successor = next(iter(successors))
-            successor_key = nncf_graph.get_node_key_by_id(successor.node_id)
-            if successor.metatype in ELEMENTWISE_LAYER_METATYPES and len(nncf_graph.get_previous_nodes(successor)) == 1:
-                preprocessing_nodes.append(successor)
-                return self._get_input_preprocessing_nodes(nncf_graph, successor_key, preprocessing_nodes)
-        return preprocessing_nodes
+    def _get_quantizer_setup(self, nncf_graph: NNCFGraph,
+                             quantizable_weighted_layer_nodes: List[QuantizableWeightedLayerNode],
+                             custom_layer_node_names: List[NNCFNodeName]) \
+            -> SingleConfigQuantizerSetup:
+        ip_graph = InsertionPointGraph(nncf_graph,
+                                       [qn.node.node_name for qn in quantizable_weighted_layer_nodes])
+
+        pattern = TF_HW_FUSED_PATTERNS.get_full_pattern_graph()
+        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
+
+        input_preprocessing_nodes = self._get_input_preprocessing_nodes(nncf_graph)
+        input_preprocessing_node_names = [n.node_name for n in input_preprocessing_nodes]
+        ignored_scopes_for_solver = self.ignored_scopes_per_group[QuantizerGroup.ACTIVATIONS] + \
+                                    input_preprocessing_node_names + custom_layer_node_names
+
+        solver = QuantizerPropagationSolver(
+            ignored_scopes=ignored_scopes_for_solver,
+            target_scopes=self.target_scopes_per_group[QuantizerGroup.ACTIVATIONS],
```

Comment:
I have got error for the classification sample with command line arguments:
--config ./examples/tensorflow/classification/configs/quantization/mobilenet_v2_imagenet_int8.json
--data /mnt/ssd/datasets/ImageNet2012/tfrecord
--mode train
--dataset-type tfrecords
```
Traceback (most recent call last):
  File "/home/alexsu/work/projects/algo/nncf_tf/source/nncf-tf/github/nncf/examples/tensorflow/classification/main.py", line 178, in run
    compression_ctrl, compress_model = create_compressed_model(model, nncf_config, compression_state)
  File "/home/alexsu/work/projects/algo/nncf_tf/source/nncf-tf/github/nncf/nncf/tensorflow/helpers/model_creation.py", line 87, in create_compressed_model
    compressed_model = builder.apply_to(model)
  File "/home/alexsu/work/projects/algo/nncf_tf/source/nncf-tf/github/nncf/nncf/tensorflow/api/compression.py", line 54, in apply_to
    transformation_layout = self.get_transformation_layout(model)
  File "/home/alexsu/work/projects/algo/nncf_tf/source/nncf-tf/github/nncf/nncf/tensorflow/quantization/algorithm.py", line 244, in get_transformation_layout
    custom_layer_nodes)
  File "/home/alexsu/work/projects/algo/nncf_tf/source/nncf-tf/github/nncf/nncf/tensorflow/quantization/algorithm.py", line 380, in _get_quantizer_setup
    target_scopes=self.target_scopes_per_group[QuantizerGroup.ACTIVATIONS],
KeyError: <QuantizerGroup.ACTIVATIONS: 'activations'>
```

Reply:
Fixed